lib/srv/transport/transportv1/transport.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transportv1
+
+import (
+	"context"
+	"net"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+
+	proxyv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/proxy/v1"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/teleagent"
+	"github.com/gravitational/teleport/lib/utils"
+	grpcutils "github.com/gravitational/teleport/lib/utils/grpc"
+)
+
+// Dialer is the interface that groups basic dialing methods.
+type Dialer interface {
+	DialSite(ctx context.Context, clusterName string) (net.Conn, error)
```

Comment:
why is it called dialSite? I saw that naming already in teleport codebase but never understand why it's named site